lib/samples/show_device_location/show_device_location_sample.dart

```diff
@@ -53,124 +63,160 @@ class _ShowDeviceLocationSampleState extends State<ShowDeviceLocationSample>
+                    onPressed: () => setState(() => _settingsVisible = true),
+                    child: const Text('Location Settings'),
                   ),
-                ],
+                ),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            Visibility(
+              visible: !_ready,
+              child: SizedBox.expand(
+                child: Container(
+                  color: Colors.white30,
+                  child: const Center(child: CircularProgressIndicator()),
+                ),
               ),
-            )
+            ),
           ],
         ),
       ),
+      // The Settings bottom sheet.
+      bottomSheet: _settingsVisible ? buildSettings(context) : null,
     );
   }
 
-  void onMapViewReady() async {
-    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISImageryStandard);
-
-    map.initialViewpoint = Viewpoint.fromCenter(
-      ArcGISPoint(
-        x: -110.8258,
-        y: 32.154089,
-        spatialReference: SpatialReference.wgs84,
+  // The build method for the Geometry Settings bottom sheet.
+  Widget buildSettings(BuildContext context) {
+    return Container(
+      color: Colors.white,
+      padding: EdgeInsets.fromLTRB(
+        20.0,
+        0.0,
+        20.0,
+        max(
+          20.0,
+          View.of(context).viewPadding.bottom /
+              View.of(context).devicePixelRatio,
+        ),
+      ),
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Row(
+            children: [
+              Text(
+                'Location Settings',
+                style: Theme.of(context).textTheme.titleLarge,
+              ),
+              const Spacer(),
+              IconButton(
+                icon: const Icon(Icons.close),
+                onPressed: () => setState(() => _settingsVisible = false),
+              ),
+            ],
+          ),
+          Row(
+            children: [
+              const Text('Show Location'),
+              const Spacer(),
+              // A switch to start and stop the location data source.
+              Switch(
+                value: (_status == LocationDataSourceStatus.started),
+                onChanged: (_) {
+                  if (_status == LocationDataSourceStatus.started) {
+                    _mapViewController.locationDisplay.stop();
+                  } else {
+                    _mapViewController.locationDisplay.start();
+                  }
+                },
+              ),
+            ],
+          ),
+          Row(
+            children: [
+              const Text('Auto-Pan Mode'),
+              const Spacer(),
+              // A dropdown button to select the auto-pan mode.
+              DropdownButton(
+                value: _autoPanMode,
+                onChanged: (value) {
+                  _mapViewController.locationDisplay.autoPanMode = value!;
```

Comment:
AI suggests changes the line to 
```dart
_mapViewController.locationDisplay.autoPanMode = value as LocationDisplayAutoPanMode;
```

Reply:
`value` has the type `LocationDisplayAutoPanMode?`. Using `as LocationDisplayAutoPanMode` does the same thing as `!`, but with a lot more characters. I can't see any benefit.